GVRf/Framework/src/org/gearvrf/scene_objects/GVRTextViewSceneObject.java

```diff
@@ -0,0 +1,310 @@
+                text);
+    }
+
+    /**
+     * Shows a {@link TextView} on a {@linkplain GVRSceneObject scene object}
+     * with both view's default height and width and quad's default height and
+     * width.
+     * 
+     * @param gvrContext
+     *            current {@link GVRContext}
+     * @param gvrActivity
+     *            a {@link GVRActivity}
+     * @param text
+     *            {@link String} to show on the textView
+     */
+
+    public GVRTextViewSceneObject(GVRContext gvrContext,
+            GVRActivity gvrActivity, String text) {
+        this(gvrContext, gvrActivity, DEFAULT_QUAD_WIDTH, DEFAULT_QUAD_HEIGHT,
+                text);
+    }
+
+    /**
+     * Shows a {@link TextView} on a {@linkplain GVRSceneObject scene object}
+     * with both view's default height and width and quad's default height and
+     * width. And the default text of this TextView will be "Hello World!"
+     * 
+     * @param gvrContext
+     *            current {@link GVRContext}
+     * @param gvrActivity
+     *            a {@link GVRActivity}
+     */
+    public GVRTextViewSceneObject(GVRContext gvrContext, GVRActivity gvrActivity) {
+        this(gvrContext, gvrActivity, "Hello World!");
+    }
+
+    /**
+     * Set the text size.
+     * 
+     * @param newSize
+     *            The new text size.
+     */
+    public void setTextSize(float size) {
+        mTextView.setTextSize(size);
+    }
+
+    /**
+     * Get the current text size.
+     * 
+     * @return The current text size.
+     */
+    public float getTextSize() {
+        return mTextView.getTextSize();
+    }
+
+    /**
+     * Set the text color.
+     * 
+     * @param color
+     *            The text color, in Android {@link Color} format. The
+     *            {@linkplain Color#alpha(int) alpha component} is ignored.
+     */
+    public void setTextColor(int color) {
+        mTextView.setTextColor(color);
+    }
+
+    /**
+     * Set the text string to be displayed.
+     * 
+     * @param string
+     *            The new text string to be displayed.
+     */
+    public void setText(String str) {
+        mTextView.setText(str);
+    }
+
+    /**
+     * Set the view's background color.
+     * 
+     * @param color
+     *            The view's background color, in Android {@link Color} format.
+     *            The {@linkplain Color#alpha(int) alpha component} is ignored.
+     */
+    public void setBackgroundColor(int color) {
+        mTextViewContainer.setBackgroundColor(color);
+    }
+
+    /**
+     * Set the view's background with resource drawable.
+     * 
+     * @param drawable
+     *            The view's background.
+     */
+    public void setBackGround(Drawable drawable) {
+        mTextViewContainer.setBackground(drawable);
+    }
+
+    /**
+     * Get the current text string.
+     * 
+     * @return The current text string that is being displayed.
+     */
+    public String getText() {
```

Comment:
Should just return a `CharSequence`. If the app needs a `String`, it can call `.toString()` itself.

(Yes, I know: I'm Mr Convenience Function, Mr Don't Bloat The App. But the underlying `TextView` has good reasons to return a `CharSequence` and not a `String`, and we should probably follow its lead. There's nothing wrong with adding

    public String getTextString() { return getText().asString(); }